### PR TITLE
Amigara mine jsonify

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
Content "Amigara mine jsonify"

#### Purpose of change
Continuation of ongoing project of dehardcoding mine mapgen.

#### Describe the solution
- Added pump to mine entrance to drain water flooding the lower levels of mine. Obviously, the pump doesn't work anymore, so there are puddles of water everywhere in mine tunnels.
- Updated mine shaft middle and lower levels.
- Updated overmap special definitions for every type of jsonized mines: added missing roof over the loading zone.
- Added railroad ramps. Based on road ramps. _Pushing_ the trolley works as expected. _Pulling_ the trolley makes player lose the grab of it after z-level change.
- Lowered time to trigger amigara horror spawn from 1 minute to 10 seconds.
- Added `AMIGARA_WHISPERS` timed event which will trigger if player has read the logs. If there is at least one `odd fault` terrain on level, player will hear strange/creepy whispers from poor souls of the faults. This whispers will last for 5 minutes.
-  Updated amigara horror's fear paralyse attack so it now require a direct line of vision to its target.
- Moved amigara-related snippets to their own respective file.
- There are no monsters in the whole underground location (save for amigara horrors) in accordance to the logs, where it was revealed that miners were arrested (and most probably moved to somewhere else).
- The loot is rather scarce too. Rarely mine materials and rarely mine equipment on mine shaft lower level and mine tunnels level. In mine finale there are two closed crates with archaeologist's tools. And the reward for killing the horrors, of course.
- Created `amigara` monster faction and assigned amigara horrors to it so they don't fight each other just after the spawn.
- Removed hardcoded mapgen.

#### Describe alternatives you've considered
None.

#### Testing
Debug-spawned location, checked if everything is ok. Walked on foot to the mine finale. Pushed the trolley to the mine finale.

#### Additional context
New (non-working) facility for draining the water from lower levels
изображение

Bird-eye view of mine shaft lower level and its continuation. Note the puddles of water slowly flooding the level. Also note railroad ramps in the end of the tracks
изображение

Bird-eye view of mine tunnels. Note the sinkholes dotting the location.
изображение

Mine finale with two crates of archaeologist's tools, computer and a faultline.
изображение

### CODE REVIEW

**Comment by actual-nh** [comment]
> Created amigara monster faction and assigned amigara horrors to it so they don't fight each other just after the spawn.

Quite. Why NEUTRAL toward each other and not the default FRIENDLY?

File: data/json/snippets/amigara.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    ]
+  },
+  {
+    "type": "snippet",
+    "category": "amigara2",
+    "text": [
+      "ENTRY 49:\nWe've stopped mining operations in this area, obviously, until archaeologists have the chance to inspect the area.  This is going to set our schedule back by at least a week.  This stupid artifact-preservation law has been in place for 50 years, and hasn't even been up for termination despite the fact that these mining operations are the backbone of our economy.\nENTRY 52:\nStill waiting on the archaeologists.  We've done a little light inspection of the faultline; our sounding equipment is insufficient to measure the depth of the concavities.  The equipment is rated at 15 miles depth, but it isn't made for such narrow tunnels, so it's hard to say exactly how far back they go."
```
Comment (actual-nh) [inline data/json/snippets/amigara.json:13]:
```suggestion
      "ENTRY 49:\nWe've stopped mining operations in this area, obviously, until archaeologists have the chance to inspect the area.  This is going to set our schedule back by at least a week.  This stupid artifact-preservation law has been in place for 50 years, and hasn't even been up for termination despite that these mining operations are the backbone of our economy.\nENTRY 52:\nStill waiting on the archaeologists.  We've done a little light inspection of the faultline; our sounding equipment is insufficient to measure the depth of the concavities.  The equipment is rated at 15 miles depth, but it isn't made for such narrow tunnels, so it's hard to say exactly how far back they go."
```

File: data/json/snippets/amigara.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "category": "amigara3",
+    "text": [
+      "ENTRY 54:\nI noticed a couple of the guys down in the chamber with a chisel, breaking off a piece of the sheer wall.  I'm looking the other way.  It's not like the eggheads are going to notice a little piece missing.  Fuck em.\nENTRY 55:\nWell, the archaeologists are down there now with a couple of the boys as guides.  They're hardly Indiana Jones types; I doubt they been below 20 feet.  I hate taking guys off assignment just to babysit the scientists, but if they get hurt we'll be shut down for god knows how long.\nENTRY 58:\nThey're bringing in ANOTHER CREW?  Christ, it's just some cave carvings!  I know that's sort of a big deal, but come on, these guys can't handle it?"
+    ]
+  },
+  {
+    "type": "snippet",
+    "category": "amigara4",
+    "text": [
+      "MINE OPERATIONS SUSPENDED; CONTROL TRANSFERRED TO AMIGARA PROJECT UNDER IMPERATIVE 2:07B.\nFAULTLINE SOUNDING HAS PLACED DEPTH AT 30.09 KM.\nDAMAGE TO FAULTLINE DISCOVERED; NEPOWER MINE CREW PLACED UNDER ARREST FOR VIOLATION OF REGULATION 87.08 AND TRANSFERRED TO LAB 89-C FOR USE AS SUBJECTS.\nQUALITY OF FAULTLINE NOT COMPROMISED.\nINITIATING STANDARD TREMOR TEST…"
```
Comment (actual-nh) [inline data/json/snippets/amigara.json:27]: "Nepower"?

File: src/timed_event.cpp [hunk 3]
```diff
@@ -275,17 +276,38 @@ void timed_event::per_turn()
...
+            bool faults = false;
+            for( const tripoint &p : here.points_on_zlevel() ) {
+                if( here.ter( p ) == t_fault ) {
+                    faults = true;
+                    break;
+                }
+            }
+
+            if( calendar::once_every( time_duration::from_seconds( 10 ) ) && faults ) {
+                add_msg( m_info, "You hear someone whishpering \"%s\"",
```
Comment (actual-nh) [inline src/timed_event.cpp:301]:
```suggestion
                add_msg( m_info, "You hear someone whispering \"%s\"",
```

**Comment by Night-Pryanik** [comment]
> Why NEUTRAL toward each other and not the default FRIENDLY? (I definitely like, BTW.)

I guess they were different people prior to converting to horrors, and they shouldn't necessarily be friendly to each other.

I could convert to FRIENDLY though, if needed.

**Comment by actual-nh** [comment]
> > Why NEUTRAL toward each other and not the default FRIENDLY? (I definitely like, BTW.)
> 
> I guess they were different people prior to converting to horrors, and they shouldn't necessarily be friendly to each other.

Point.
 
> I could convert to FRIENDLY though, if needed.

I can see it either way. (I particularly noticed because I've put up a faction-altering PR - #48077 - and at Venera3's request it includes an "amigara" faction that, as is the default for own-faction members, has them as FRIENDLY to each other. I was wondering if there was something in their backstory that should make them NEUTRAL to each other.)

**Comment by Night-Pryanik** [comment]
@someone_at_github I'd like to receive a notification when someone force-pushes changes to my PRs. I opened this PR only to accidentally discover that now it's a mess of commits I have to rake.

**Comment by Night-Pryanik** [comment]
Damn, looks like I accidentally merged the PR. I didn't want to, honestly!

**Pull request merged**